include/roboteam_utils/containers/circular_buffer.h

```diff
@@ -40,7 +40,7 @@ namespace rtt::collections {
 
     public:
         //Default constructor
-        circular_buffer() : _current_offset(0), _current_size(0) {
+        constexpr circular_buffer() : _current_offset(0), _current_size(0) {
```

Comment:
`= default;`

Reply:
No? Default initialization will not initalize both both integer member variables (_current_offset, _current_size)

Reply:
Nvm I was wrong

Reply:
This even gives an error if I try to compile it because it's not constexpr :). Guess making things constexpr is a good practice to prevent accidental UB from slipping in.

Reply:
Nah it would set it to `0` because of class initialization rules, but this is fine.

Reply:
https://godbolt.org/z/Gfbc5E

There's a pretty clear difference in behavior, one compiles other doesn't. Also, one clearly explicitly returns zero whilst the other one does not (UB).

Reply:
Yes, you should initialize it using {}, not leave it uninitialized.